fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/model/DataModelVisitor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.ibm.fhir.database.utils.model;
+
+
+/**
+ * Visitor interface used to traverse the PhysicalDataModel
+ */
+public interface DataModelVisitor {
+
+    /**
+     * Process a foreign key constraint from the model
+     * @param fk
+     */
+    public void visited(Table fromChildTable, ForeignKeyConstraint fk);
```

Comment:
curious, why using "visited" instead of "visit"?

Reply:
It's on the receiving side. It's just to differentiate from the visit(...) call on the model object itself (e.g. Table) which is the thing being visited. The DataModelVisitor goes to visit(...) each object in the PhysicalDataModel, and each object then tells the visitor, "hey, you've visited(...) me". That's the logic I use in my head.